code/stats/scoring.cpp

```diff
@@ -130,6 +207,75 @@ void parse_rank_tbl()
 	}
 }
 
+void sort_ranks()
+{
+	bool shouldSort = false;
+
+	// be sure no ranks have equal point values
+	for (int i = 0; i < (int)Ranks.size(); i++) {
+		for (int j = (i + 1); j < (int)Ranks.size(); j++) {
+			if (Ranks[i].points == Ranks[j].points) {
+				Ranks[j].points++;
+				shouldSort = true;
+				mprintf(("Rank %s and %s are equal! Increasing %s to %i\n",
+					Ranks[i].name,
+					Ranks[j].name,
+					Ranks[j].name,
+					Ranks[j].points));
+			}
+		}
+	}
```

Comment:
This is insufficient.
Assume you add ranks with points 2, 1 and 1.
It'll check the 2 with both 1's and be okay.
Then it'll check the first 1 and increase the second 1 to 2, but that'll never be checked against the first 2 again.
I recommend just erroring on two identical scores.

Reply:
Hmm. I thought I checked that case specifically but it's been months since. I'll give it all another look. I've learned a lot since, too.

Reply:
Yeah, I don't know what I was thinking when I wrote that. Good catch, though. I think this is probably solvable, but yeah let's go path of least resistance. Error applied.